### PR TITLE
ENH: add append to RectPartition, closes #351

### LINKED ISSUE #351: RectPartition.extend
grid and intervalprod has this method, partitions should as well. It is a simple addition, it simply wraps:

```python
def extend(partition):
    return self.insert(self.ndim, partition)
```

### PR DESCRIPTION
**Pull request by adler-j**
Issue was wrongly named, should be `append`. Added this.

Also added a shorthand to `uniform_partition` in `RectPartition.__repr__`.

### CODE REVIEW

File: odl/discr/grid.py [hunk 1]
```diff
@@ -520,9 +520,8 @@ def append(self, other):
 
         Parameters
         ----------
-        other : `IntervalProd`, `float` or `array-like`
-            The set to be inserted. A `float` or array a is
-            treated as an ``IntervalProd(a, a)``.
+        other : `TensorGrid`, `float` or `array-like`
```
Comment (kohr-h) [inline odl/discr/grid.py]: Don't quite remember. Didn't we settle for not accepting arrays and floats?

File: odl/discr/grid.py [hunk 2]
```diff
@@ -520,9 +520,8 @@ def append(self, other):
...
+            The set to be inserted.
```
Comment (kohr-h) [inline odl/discr/grid.py]: no dot

File: odl/discr/partition.py [hunk 3]
```diff
@@ -378,19 +378,60 @@ def insert(self, index, other):
 
         Examples
         --------
+        >>> part1 = uniform_partition([0, -1], [1, 2], (3, 3))
+        >>> part2 = uniform_partition(0, 1, 5)
+        >>> part1.insert(2, part2)
+        uniform_partition([0.0, -1.0, 0.0], [1.0, 2.0, 1.0], [3, 3, 5])
```
Comment (kohr-h) [inline odl/discr/partition.py]: Perhaps more instructive to insert in the middle?

File: odl/discr/partition.py [hunk 4]
```diff
@@ -378,19 +378,60 @@ def insert(self, index, other):
...
+        Parameters
+        ----------
+        other : `RectPartition`,
+            The set to be inserted.
+
+        Examples
+        --------
+        >>> part1 = uniform_partition([0, -1], [1, 2], (3, 3))
+        >>> part2 = uniform_partition(0, 1, 5)
+        >>> part1.insert(2, part2)
```
Comment (kohr-h) [inline odl/discr/partition.py]: use append

**Comment by adler-j** [comment]
Updated by comments, merge?

**Pull request merged**